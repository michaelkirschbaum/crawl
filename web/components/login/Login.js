import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { setAuthentication } from '../app/appActions'
import './Login.css'

const mapDispatchToProps = dispatch => {
  return {
    setAuthentication: status => {
      dispatch(setAuthentication(status))
    }
  }
}

class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      first: '',
      last: '',
      redirect: false
    }
    this.onChange = this.onChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.authenticate = this.authenticate.bind(this)
  }

  onChange(event) {
    this.setState({[event.target.name]: event.target.value})
  }

  handleSubmit() {
    const { first, last } = this.state

    this.authenticate(first, last)
      .then(() => this.setState({ redirect: true }))
      .catch(err => console.log(err))
  }

  authenticate(first, last) {
    return new Promise (
      function (resolve, reject) {
        if (first && last)
          resolve(first)
        else
          reject(new Error('authentication failed'))
      }
    )
  }

  render() {
    var { first, last, redirect } = this.state
    const from = { pathname: '/' }

    if (redirect)
      return <Redirect to={from}/>

    return (
      <div className="Login">
        <h1>Login</h1>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="first" onChange={this.onChange} placeholder="First name"/><br />
          <input type="text" name="last" onChange={this.onChange} placeholder="Last name"/><br />
          <input type="submit" value="Login"/>
        </form>
      </div>
    )
  }
}

export default connect(null, mapDispatchToProps)(Login)
