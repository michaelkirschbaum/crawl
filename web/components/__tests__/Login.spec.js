import React from 'react'
import { Login } from '../../components/login/Login'
import { shallow } from 'enzyme'

describe('Login', () => {
  it('should render correctly', () => {
    const component = shallow(<Login location={{ state: { pathname: '/'} }}/>)
    expect(component).toMatchSnapshot()
  })
})
