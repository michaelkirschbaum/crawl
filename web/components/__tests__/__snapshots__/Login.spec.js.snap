// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    location={
      Object {
        "state": Object {
          "pathname": "/",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Login
        </h1>,
        <form
          onSubmit={[Function]}
        >
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <br />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <br />
          <input
            type="submit"
            value="Login"
          />
        </form>,
        <RedirectButton
          route="/register"
        />,
      ],
      "className": "Login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <br />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />,
            <br />,
            <input
              type="submit"
              value="Login"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Login",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "route": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Login
          </h1>,
          <form
            onSubmit={[Function]}
          >
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <br />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <br />
            <input
              type="submit"
              value="Login"
            />
          </form>,
          <RedirectButton
            route="/register"
          />,
        ],
        "className": "Login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
              />,
              <br />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />,
              <br />,
              <input
                type="submit"
                value="Login"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "route": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
